docs/glossary.md

```diff
@@ -0,0 +1,171 @@
+# Glossary
+
+This glossary serves as a foundation for having a common wording and language.
+The motivation behind this document was the many different namings and, hence,
+the resulting communication issues when talking about schema composition.
+
+### Apollo Federated Graph ([Link](https://www.apollographql.com/docs/federation/))
+
+The process of composing multiple Apollo Subgraphs into one schema, usually via
+a Gateway. An implementation of Schema composition. The result of composing
+multiple Apollo federation Subgraphs is a Apollo Federated Graph. The Apollo
+Federated Graph is usually exposed via an Apollo Federation Gateway
+(apollo/gateway; apollo-router) or Apollo Federation compatible gateway (e.g.
+graphql-mesh).
+
+### Apollo Federation
+
+An implementation of **Subschema Directive-based Schema compositions** that is
+defined by the Apollo Federation Specification. GraphQL schemas (**Apollo
+Federation Subgraph**s) are composed of a single GraphQL schema (**Apollo
+Federated Graph**) on a Gateway level (**Apollo Federation Gateway**).
+
+### Apollo Federation Gateway
+
+A gateway that usually uses a schema registry (e.g. Apollo Studio or GraphQL
+Hive) for fetching the apollo subgraph configurations. It is the entry point for
+incoming GraphQL requests. It analyzes the incoming operations, builds a query
+plan in order to find the quickest way of retrieving all the data from the
+apollo federation subgraphs, and then executes that plan.
+
+The approach of combining multiple GraphQL services into a unified GraphQL Graph
+on a Gateway using meta-information applied on the individual GraphQL services
+schema type definitions.
+
+### Apollo Federation Subgraph ([Link](https://www.apollographql.com/docs/federation/subgraphs/))
+
+A service that exposes a GraphQL schema that is consumed by a gateway in order
+to create a federated graph. The services schema uses directive annotations on
+types and fields as a hint for a gateway on how it should combine the services
+schema with the schema of the other Apollo federation subgraphs. A Apollo
+Federation Subgraph is a GraphQL schema with Apollo Federation schema
+directives. Also See **Subschema**.
+
+### Apollo Graph Platform
+
+Branding used by Apollo for describing their open-source products.
+
+> The leading open-source GraphQL implementation with 17M downloads per month
+> and the only end-to-end cloud management solution for GraphQL. Source:
+> [https://www.apollographql.com/](https://www.apollographql.com/) (2022/02/24)
+
+### Apollo Studio
+
+A schema registry for registering and validating Apollo federation subgraphs.
+
+### Data Source
+
+A “dumb” service or database that exposes raw data. Often a data source is
+consumed by a service, for reading and/or persisting business logic state.
+
+It is hard to define a border between Data Source and Service. E.g. a PostgreSQL
```

Comment:
Isn't the difference between Source and Service the presence of business logic?

Reply:
Maybe _data source_ is not the best term here. A data source does not directly refer to the persistence layer of data, so I can source data from a DB via an API/Interface that also include embedded business logic; the data source is the API, not the DB. 
Nonetheless, it's also true that a view on a database can include a business logic. A query can include business logic as well, unless it's a `select * from ...`. The example of the trigger is quite correct.